js/plates-model/fields-collision.js

```diff
@@ -0,0 +1,60 @@
+import Subduction from './subduction'
+import Orogeny from './orogeny'
+import VolcanicActivity from './volcanic-activity'
+
+function applyDragForces (bottomField, topField) {
+  bottomField.draggingPlate = topField.plate
+  topField.draggingPlate = bottomField.plate
+}
+
+function subduction (bottomField, topField) {
+  if (!bottomField.subduction) {
+    bottomField.subduction = new Subduction(bottomField)
+  }
+  if (!topField.volcanicAct) {
+    topField.volcanicAct = new VolcanicActivity(topField)
+  }
+  bottomField.subduction.setCollision(topField)
+  topField.volcanicAct.setCollision(bottomField)
+}
+
+function islandCollision (bottomField, topField) {
+  // Island collision (it will be merged with colliding plate).
+  topField.plate.mergeIsland(bottomField, topField)
+}
+
+function orogeny (bottomField, topField) {
+  applyDragForces(bottomField, topField)
+  if (!bottomField.orogeny) {
+    bottomField.orogeny = new Orogeny(bottomField)
+  }
+  if (!topField.orogeny) {
+    topField.orogeny = new Orogeny(topField)
+  }
+  bottomField.orogeny.setCollision(topField)
+  topField.orogeny.setCollision(bottomField)
+}
+
+export default function fieldsCollision (bottomField, topField) {
+  if (bottomField.boundary && topField.boundary) {
+    // Skip collision between field at boundary, so simulation looks a bit cleaner.
+    return
+  }
+
+  if (bottomField.isOcean) {
+    subduction(bottomField, topField)
+    if (bottomField.isContinentBuffer && topField.isOcean) {
+      // Special case when the continent is "trying" to subduct under the ocean. Apply drag force to stop both plates.
+      applyDragForces(bottomField, topField)
+    }
+  } else if (bottomField.isContinent) {
+    if (topField.isContinent) {
+      orogeny(bottomField, topField)
+    } else { // topField.isOcean || topField.isIsland
```

Comment:
It seems like this `else` case could almost be merged with the `bottomField.isContinentBuffer && topField.isOcean` case above. If the condition was instead `(bottomField.isContinentBuffer || bottomField.isContinent) && (topField.isOcean || topField.isIsland)`, the comment in both blocks wouldn't need to be duplicated. The only change is that this would create drag forces when a continent buffer collides with an island, which seems like it could be a desirable change?

Reply:
> If the condition was instead (bottomField.isContinentBuffer || bottomField.isContinent) && (topField.isOcean || topField.isIsland)

Not sure where would we put it? Note that `if`s are now nested. The first level checks bottom field and it's very simple, no && and ||. Then we look at the top field. I find it a bit easier to reason about.

> The only change is that this would create drag forces when a continent buffer collides with an island, which seems like it could be a desirable change?

Yes, that makes sense. It'll be consistent with the logic related to orogeny (continent vs continent => orogeny, contient / cont. buffer vs non-continent => only drag force).